sequencer/src/block2.rs

```diff
@@ -99,7 +111,7 @@ impl BlockPayload {
     fn get_tx_table_len_proof(&self, vid: &impl PayloadProver<RangeProof>) -> Option<&RangeProof> {
         self.tx_table_len_proof
             .get_or_init(|| {
-                vid.payload_proof(&self.payload, 0..TxTableEntry::byte_len())
+                vid.payload_proof(&self.payload, self.tx_table_len_range())
```

Comment:
I think we need to change the proof verification to require that if this range is less than `TxTableEntry::byte_len`, then the payload length equals the range length.

Reply:
By "proof verification" I guess you mean `TxInclusionProof::verify()`, right? Perhaps you're referring to the following line in `TxInclusionProof::verify()`?

https://github.com/EspressoSystems/espresso-sequencer/blob/e95618c607ce3228f3b4bad60abe882f78741693/sequencer/src/block2.rs#L323-L323

If the payload is that small then it must contain zero txs, and any purported tx inclusion proof will fail anyway (as it should). We do have some tests which confirm `transaction_with_proof()` fails to create a proof for such small blocks.

But we are missing a test for maliciously-generated `TxInclusionProof`s. I do think such a proof would be correctly rejected by current code, even without your suggestion. But I probably should add a test for this anyway.

Reply:
Ah, you're right...there is no valid inclusion proof in this case. And it does seem that there is nothing a malicious prover can do to convince a verifier otherwise, since the verifier can always see that the length of the payload is too short. Still a good test to add indeed